### ISSUE TITLE
datashader requires `bokeh` on conda-forge ?

### ISSUE THREAD

**Issue by raphaelquast** [holoviz/datashader#1053]
#### Description of expected behavior and the observed behavior
I've noticed that datashader on `conda-forge` has a dependency on `bokeh` which I think is quite odd.
(It's one of the biggest dependencies and if one intends to use `matplotlib`, it is obsolete)

#### Complete, minimal, self-contained example code that reproduces the issue

using the following in a fresh environment 
```
conda install -c conda-forge datashader
```
adds the following dependency:
```
bokeh                                 2.4.2  py39hcbf5309_0      conda-forge/win-64      14 MB
```

... also with `mamba` a query of the dependencies shows:
`mamba repoquery whoneeds --tree bokeh`
```
bokeh[2.4.2]
  └─ dask[2022.1.1]
     └─ datashader[0.13.0]
```

**Comment by jbednar** [comment]
Right. There's an old bokeh_ext.py file in Datashader that's deprecated, and when that gets removed the Bokeh dependency should be removed as well. Not sure when we'll get to that cleanup task, but if someone wants to volunteer, it could happen in the next release (which is due in the next few weeks).

**Comment by raphaelquast** [comment]
Thanks for the quick response!
Unfortunately I don't have any insights into datashader-internals so I'm not the right one to volunteer for this...
(except it's just about opening a pull-request to delete that file...)

**Comment by jbednar** [comment]
I think all that's required is to delete `bokeh_ext.py` and `test_bokeh_ext.py`. Looks like `setup.py` already doesn't require bokeh or any plotting library, so separately the conda-forge recipe will need updating if it's requiring bokeh. I don't actually see where in https://github.com/conda-forge/datashader-feedstock/blob/main/recipe/meta.yaml that requirement would come from; I only see bokeh-related things under the tests section.

**Comment by raphaelquast** [comment]
hey, so general question... there's also a `mpl_ext.py` but that does not result in a dependency on `matplotlib` (as long as you don't import it) so as I understand it, wherever this dependency is coming from, deleting `bokeh_ext.py` and `test_bokeh_ext.py` should not resolve this...

I had a quick look and now I think that this whole problem is actually coming from dask...
(I see even less reason for dask to add bokeh as a general dependency...)

from their meta.yml:  (https://github.com/conda-forge/dask-feedstock/blob/main/recipe/meta.yaml)

```
requirements:
  host:
    - python >=3.8
  run:
    - python >=3.8
    - dask-core {{ version }}
    - distributed {{ version }}
    - cytoolz >=0.8.2
    - numpy >=1.18
    - pandas >=1.0
    - bokeh >=2.4.2
    - jinja2
```

**Comment by raphaelquast** [comment]
hey... quick followup:  (https://github.com/dask/dask/issues/8838#issuecomment-1077673339)

> "You probably want the dask-core package"

so I guess the actual solution for this is to replace
```
requirements
    - dask >=0.18.0
```
with (not sure about ther version-numbers...)
```
requirements
    - dask-core >=0.18.0
```

**Comment by jbednar** [comment]
Sounds great! I'd add one more to-do, which is to add dask to the examples requirements in setup.py now that the core requirement will only be dask-core; that way people can run dask-based examples.

**Comment by jbednar** [comment]
Based on the comments in setup.py:

```
    # `conda install dask[complete]` happily gives you dask...which is
    # happily like pip's dask[complete]. (conda's dask-core is more
    # like pip's dask.)
    'dask[complete] >=0.18.0',
```

we'll have to do some testing before simply switching dask for dask-core. I.e., we build both conda and pip packages, and we need some specification that works on both platforms. It sounds like the current one works for our purposes for both pip and conda, but that we may run into problems (or else why would `complete` have been there?) if we remove that. Good to do, but not something I'll have time to investigate soon.

**Comment by raphaelquast** [comment]
awesome! thanks for fixing this 🥳

**Comment by ianthomas23** [comment]
You're welcome! This fix isn't in the latest release 0.14.2 on PyPI, although it was supposed to be. It will definitely be in 0.14.3 which will hopefully be out in a couple of weeks.

**Comment by raphaelquast** [comment]
Great! No need for speed on my side... just glad it is going to come soon! (It was a long-term goal on my side to get rid of `bokeh` as a dependency for [EOmaps ](https://github.com/raphaelquast/EOmaps) which uses `datashader` for `matplotlib` plots.)
